workspaces/marketplace/packages/cli/src/commands/export-csv/index.ts

```diff
@@ -0,0 +1,273 @@
+
+export default async ({
+  outputFile,
+  pluginsYamlPath,
+  recursive,
+}: OptionValues) => {
+  if (!pluginsYamlPath) {
+    console.error(
+      'No plugins path provided! Provide one using --plugins-yaml-path',
+    );
+    process.exit(1);
+  }
+
+  /** Parse the YAML files into entities */
+  const yamls = parseYamls(pluginsYamlPath.split(','), recursive);
+
+  /** The headers we want to include in the CSV */
+  const headers = [
+    'type',
+    'name',
+    'title',
+    'description',
+    'author',
+    'categories',
+    'lifecycle',
+    'packages',
+    'role',
+    'backend plugins',
+    'frontend plugins',
+  ];
+
+  /** A hashmap of packages for the plugins CSV generator */
+  const packages: Record<string, MarketplacePackage> = {};
+
+  /** The generator for backstage marketplace Packages */
+  const packageCSV = new CSVGenerator<MarketplacePackage>(
+    getColumns<MarketplacePackage>(headers, [
+      () => 'Package',
+      yaml => yaml?.metadata?.name,
+      yaml => yaml?.metadata?.title,
+      () => undefined, // packages don't have a description in their type
+      yaml =>
+        yaml?.spec?.developer ||
+        yaml?.spec?.author ||
+        yaml?.spec?.owner ||
+        ((yaml?.spec?.authors as string[]) || []).join(','),
+      () => undefined, // packages don't have categories in their type
+      yaml => yaml?.spec?.lifecycle,
+      yaml => (yaml?.spec?.partOf || []).join(', '),
+      yaml => yaml?.spec?.role || yaml?.spec?.backstage?.role,
+      () => undefined, // packages don't have backend plugins in their type
+      () => undefined, // packages don't have frontend plugins in their type
+    ]),
+  );
+
+  /** The generator for backstage marketplace Plugins */
+  const pluginCSV = new CSVGenerator<MarketplacePlugin>(
+    getColumns<MarketplacePlugin>(headers, [
+      () => 'Plugin',
+      p => p?.metadata?.name,
+      p => p?.metadata?.title,
+      p => p?.metadata?.description,
+      p =>
+        p?.spec?.developer ||
+        p?.spec?.author ||
+        p?.spec?.owner ||
+        (p?.spec?.authors || []).join(','),
+      p => ((p?.spec?.categories as string[]) || []).join(', '),
+      p => p?.spec?.lifecycle,
+      p => (p?.spec?.packages || []).join(', '),
+      () => undefined, // plugins don't have a role in their type
+      p =>
+        getPackagesOfType(p?.spec?.packages || [], packages, [
+          'backend-plugin',
+          'backend-plugin-module',
+        ]),
+      p =>
+        getPackagesOfType(p?.spec?.packages || [], packages, [
+          'frontend-plugin',
+        ]),
+    ]),
+  );
+
+  /** Process each YAML file */
+  for (const yaml of await yamls) {
+    if (isMarketplacePackage(yaml)) {
+      packages[yaml.metadata.name] = yaml;
+      packageCSV.addRow(yaml);
+    } else if (isMarketplacePlugin(yaml)) {
+      pluginCSV.addRow(yaml);
+    }
+  }
+
+  /** Generate the final CSV */
+  const finalCSV = `${packageCSV.generate()}${pluginCSV.generate(false)}`;
```

Comment:
Okay. Two different "tables" in the same CSV doesn't look right to me.

Can you change that to have a merged table as we discussed.

With columns for

* Frontend package
* Frontend version
* Backend package
* Backend version
* Other packages